### ISSUE TITLE
[Bug] Download speed is vastly different between manga

### ISSUE THREAD

**Issue by franzbuenaventura** [CarlosEsco/Neko#212]
### Device information
* Neko version: 2.0.10
* Android phone & version: V40 & Android 9.0

## Steps to reproduce
1. Download Chapters in "Kimi wa 008" or "Berserk of Gluttony" 
2. Download Yankee JK KuzuHana-Chan (Tested downloading Ch. 9)
3. Compare speed

### Expected behavior
Downloads should be as fast as the internet is on or at least the expected speed is the same on all manga

### Actual behavior
For the first 2 manga I mention, the download is taking like 3-5 mins per page but on the 3rd manga which has 22 pages only took less than a minute to download everything. 

### Other details
The image server is set to automatic

**Comment by CarlosEsco** [comment]
few things at play here that affect speed and none of this on Neko. 

1.  image sizes
     kimi wa pages are all ~500kb
     berserk of gluttony are all ~2mb
     yankee JK are all ~300kb
2.  Dex servers and when you try to access them
     they are getting slammed,  so all it takes is a request to download 1 image to slow down to 
      cause the whole chapter download to slow.
3.  how popular is the manga, and how new the chapter is.
    If its not popular its probably or newer then its not in dex's cache.  meaning its gonna serve slower.  
    kimi wa chapters get around 3k views, yankees 9th chapter is around 10k while Berserk of Gluttony's latest are 30k+


So to summarize (ignoring the fact that i cant even add server speed to this and also affects)
- kimi wa pages are medium in size but get little views so i doubt its in the cache.
- berserk of gluttuny images are on the larger size and while they get lots of views the last chapters was 14 days ago.  So i doubt its still in cache.
- yankee has the smallest chapter, and chapters 9 was updated recently and it gets decent views.  Its definitely in cache.

**Comment by franzbuenaventura** [comment]
Thanks for taking your time explaining that @CarlosEsco. 
Quite surprise how MangaDex operates on caching and loading Manga, to be fair there are probably millions of chapters on the site.

**Comment by Efreak** [comment]
As an additional comment, I don't think this is the only issue. I've noticed that while Neko downloads two chapters simultaneously, it seems to prioritize the second chapter over the first. I can start chapters 1-10 downloading, it'll start with 1 and 2, and by the time chapter 1 finishes downloading, chapters 2-3 have finished and 4 has started. I generally just ignore it since I usually download quite a bit ahead, but occasionally it gets me; reordering downloads usually fixes it if pausing and resuming doesn't.

**Comment by CarlosEsco** [comment]
I'll have to play with it more but in a quick check the first one at some point usually gets a timeout or two getting the images which is causes they thread to wait
